geth/transactions/txqueue_manager.go

```
@@ -75,34 +77,44 @@ func (m *Manager) QueueTransaction(tx *common.QueuedTx) error {
 		to = tx.Args.To.Hex()
 	}
 	log.Info("queue a new transaction", "id", tx.ID, "from", tx.Args.From.Hex(), "to", to)
-	err := m.txQueue.Enqueue(tx)
+	if err := m.txQueue.Enqueue(tx); err == queue.ErrQueuedTxExist {
+		log.Warn("Transaction was already enqueud", tx.ID)
```

Typo: `s/enqueud/enqueued`

---

Any particular reason why we don't want to notify about this error? It may be beneficial for status-react as this also indicates a bug in the UI. UI should prevent from submitting the same transaction twice.

---

i thought that warning will be enough to catch this

---

ok, lets leave things how they were but with warning

---

I'm opinionated, but warning is a least useful log message :) Why not print it as an error, if we are sure that txqueue client (status-react I mean) should guarantee there is no double enqueueing?

---

actually status react cant cause this error, it just sends from, to and amount. and we generate a random uid for that

---

i would prefer just to ignore this error, as it doesn't lead to any real problem. but i will just pass it up the stack

---

Whatever makes more sense. The logic is that the decision "this condition is fine/is errorneous" should be codified. Warning level is literally asking a human to decide if it's fine or not. And human often don't know the answer :)